### DISCUSSION TITLE
RTSP Video stream not working and no error Log, but with FFMPEG only, its works.

### DISCUSSION THREAD

**Discussion by miguelangdev** [bluenviron/mediamtx#3586] [Questions]
### Question

Hello, I have an error that I can’t solve because I don’t know where the error might be. The thing is, I have a camera that generates an RTSP server, and when I connect to it with FFMPEG, it works correctly.

The point is that I want to use Mediamtx as a server, have it connect to the camera, and then connect to Mediamtx with FFMPEG. When I do this, Mediamtx shows everything as okay.

> 2024/07/30 12:29:38 INF MediaMTX v1.8.4
> 2024/07/30 12:29:38 INF configuration loaded from D:\Escritorio\Server\mediamtx.yml
> 2024/07/30 12:29:38 DEB [path camera1] created
> 2024/07/30 12:29:38 DEB path manager created
> 2024/07/30 12:29:38 INF [path camera1] [RTSP source] started
> 2024/07/30 12:29:38 DEB [path camera1] [RTSP source] connecting
> 2024/07/30 12:29:38 INF [RTSP] listener opened on :8554 (TCP), :8000 (UDP/RTP), :8001 (UDP/RTCP)
> 2024/07/30 12:29:38 INF [RTMP] listener opened on :1935
> 2024/07/30 12:29:38 INF [HLS] listener opened on :8888
> 2024/07/30 12:29:38 INF [WebRTC] listener opened on :8889 (HTTP), :8189 (ICE/UDP)
> 2024/07/30 12:29:38 INF [SRT] listener opened on :8890 (UDP)
> 2024/07/30 12:29:40 ERR [path camera1] [RTSP source] dial tcp localhost:8521: connectex: No se puede establecer una conexión ya que el equipo de destino denegó expresamente dicha conexión.
> 2024/07/30 12:29:45 DEB [path camera1] [RTSP source] connecting
> 2024/07/30 12:29:45 DEB [path camera1] [RTSP source] [c->s] OPTIONS rtsp://localhost:8521 RTSP/1.0
> CSeq: 1
> User-Agent: gortsplib
> 
> 
> 2024/07/30 12:29:45 DEB [path camera1] [RTSP source] [s->c] RTSP/1.0 401 Unauthorized
> CSeq: 1
> Public: OPTIONS,DESCRIBE,ANNOUNCE,SETUP,PLAY,PAUSE,TEARDOWN,GET_PARAMETER,SET_PARAMETER,REDIRECT
> WWW-Authenticate: Digest realm="SharpRTSPServer", nonce="975155369"
> 
> 
> 2024/07/30 12:29:45 DEB [path camera1] [RTSP source] [c->s] OPTIONS rtsp://localhost:8521 RTSP/1.0
> Authorization: Digest username="user", realm="SharpRTSPServer", nonce="975155369", uri="rtsp://localhost:8521", response="e307736f3ed4732a982bc909c40aa1f0"
> CSeq: 2
> User-Agent: gortsplib
> 
> 
> 2024/07/30 12:29:45 DEB [path camera1] [RTSP source] [s->c] RTSP/1.0 200 OK
> CSeq: 2
> Public: OPTIONS,DESCRIBE,ANNOUNCE,SETUP,PLAY,PAUSE,TEARDOWN,GET_PARAMETER,SET_PARAMETER,REDIRECT
> 
> 
> 2024/07/30 12:29:45 DEB [path camera1] [RTSP source] [c->s] DESCRIBE rtsp://localhost:8521 RTSP/1.0
> Accept: application/sdp
> Authorization: Digest username="user", realm="SharpRTSPServer", nonce="975155369", uri="rtsp://localhost:8521", response="0a193f969ebc3a50e1fcdd973693c94f"
> CSeq: 3
> User-Agent: gortsplib
> 
> 
> 2024/07/30 12:29:45 DEB [path camera1] [RTSP source] [s->c] RTSP/1.0 200 OK
> CSeq: 3
> Content-Base: rtsp://localhost:8521/
> Content-Length: 266
> Content-Type: application/sdp
> 
> v=0
> o=user 123 0 IN IP4 0.0.0.0
> s=SharpRTSP Test Camera
> m=video 0 RTP/AVP 96
> c=IN IP4 0.0.0.0
> a=control:trackID=0
> a=rtpmap:96 H264/90000
> a=fmtp:96 profile-level-id=42C033; packetization-mode=1; sprop-parameter-sets=Z0LAM9oA8AEPsBtQEBAUAAADAAQAAAMA8jxgyoA=,aM48gA==;
> 
> 2024/07/30 12:29:45 DEB [path camera1] [RTSP source] [c->s] SETUP rtsp://localhost:8521/trackID=0 RTSP/1.0
> Authorization: Digest username="user", realm="SharpRTSPServer", nonce="975155369", uri="rtsp://localhost:8521/trackID=0", response="cdd03d79989bf4bb89557ec0cf6084f2"
> CSeq: 4
> Transport: RTP/AVP;unicast;client_port=49570-49571;mode=play
> User-Agent: gortsplib
> 
> 
> 2024/07/30 12:29:45 DEB [path camera1] [RTSP source] [s->c] RTSP/1.0 200 OK
> CSeq: 4
> Session: 1
> Transport: RTP/AVP/UDP;unicast;client_port=49570-49571;server_port=50000-50001;ssrc=4321FADE
> 
> 
> 2024/07/30 12:29:45 INF [path camera1] [RTSP source] ready: 1 track (H264)
> 2024/07/30 12:29:45 DEB [path camera1] [RTSP source] [c->s] PLAY rtsp://localhost:8521/ RTSP/1.0
> Authorization: Digest username="user", realm="SharpRTSPServer", nonce="975155369", uri="rtsp://localhost:8521/", response="ac25f048efa9fcf442abcac5dc987c82"
> CSeq: 5
> Range: npt=0-
> Session: 1
> User-Agent: gortsplib
> 
> 
> 2024/07/30 12:29:45 DEB [path camera1] [RTSP source] [s->c] RTSP/1.0 200 OK
> CSeq: 5
> RTP-Info: url=rtsp://localhost:8521/;seq=1,url=rtsp://localhost:8521/;seq=1
> Range: npt=0.000-
> Session: 1
> 
> 
> 2024/07/30 12:29:51 INF [RTSP] [conn localhost:65219] opened
> 2024/07/30 12:29:51 DEB [RTSP] [conn localhost:65219] [c->s] OPTIONS rtsp://localhost:8554/camera1 RTSP/1.0
> CSeq: 1
> User-Agent: Lavf60.20.100
> 
> 
> 2024/07/30 12:29:51 DEB [RTSP] [conn localhost:65219] [s->c] RTSP/1.0 200 OK
> CSeq: 1
> Public: DESCRIBE, ANNOUNCE, SETUP, PLAY, RECORD, PAUSE, GET_PARAMETER, TEARDOWN
> Server: gortsplib
> 
> 
> 2024/07/30 12:29:51 DEB [RTSP] [conn localhost:65219] [c->s] DESCRIBE rtsp://localhost:8554/camera1 RTSP/1.0
> Accept: application/sdp
> CSeq: 2
> User-Agent: Lavf60.20.100
> 
> 
> 2024/07/30 12:29:51 DEB [RTSP] [conn localhost:65219] [s->c] RTSP/1.0 200 OK
> CSeq: 2
> Content-Base: rtsp://localhost:8554/camera1/
> Content-Length: 311
> Content-Type: application/sdp
> Server: gortsplib
> 
> v=0
> o=- 0 0 IN IP4 127.0.0.1
> s=SharpRTSP Test Camera
> c=IN IP4 0.0.0.0
> t=0 0
> m=video 0 RTP/AVP 96
> a=control:rtsp://localhost:8554/camera1/trackID=0
> a=rtpmap:96 H264/90000
> a=fmtp:96 packetization-mode=1; profile-level-id=42C033; sprop-parameter-sets=Z0LAM9oA8AEPsBtQEBAUAAADAAQAAAMA8jxgyoA=,aM48gA==
> 
> 2024/07/30 12:29:51 DEB [RTSP] [conn localhost:65219] [c->s] SETUP rtsp://localhost:8554/camera1/trackID=0 RTSP/1.0
> CSeq: 3
> Transport: RTP/AVP/UDP;unicast;client_port=31086-31087
> User-Agent: Lavf60.20.100
> 
> 
> 2024/07/30 12:29:51 INF [RTSP] [session 828bcc93] created by localhost:65219
> 2024/07/30 12:29:51 DEB [RTSP] [conn localhost:65219] [s->c] RTSP/1.0 200 OK
> CSeq: 3
> Server: gortsplib
> Session: 8856a8e0bc5a4c67864e5a478a8d7e62;timeout=60
> Transport: RTP/AVP;unicast;client_port=31086-31087;server_port=8000-8001
> 
> 
> 2024/07/30 12:29:51 DEB [RTSP] [conn localhost:65219] [c->s] PLAY rtsp://localhost:8554/camera1/ RTSP/1.0
> CSeq: 4
> Range: npt=0.000-
> Session: 8856a8e0bc5a4c67864e5a478a8d7e62
> User-Agent: Lavf60.20.100
> 
> 
> 2024/07/30 12:29:51 INF [RTSP] [session 828bcc93] is reading from path 'camera1', with UDP, 1 track (H264)
> 2024/07/30 12:29:51 DEB [RTSP] [conn localhost:65219] [s->c] RTSP/1.0 200 OK
> CSeq: 4
> RTP-Info: url=rtsp://localhost:8554/camera1/trackID=0
> Server: gortsplib
> Session: 8856a8e0bc5a4c67864e5a478a8d7e62;timeout=60
> 
> 
> 2024/07/30 12:30:15 DEB [path camera1] [RTSP source] [c->s] GET_PARAMETER rtsp://localhost:8521/ RTSP/1.0
> Authorization: Digest username="user", realm="SharpRTSPServer", nonce="975155369", uri="rtsp://localhost:8521/", response="260493a353e2989548b13e7248a00a36"
> CSeq: 6
> Session: 1
> User-Agent: gortsplib
> 
> 
> 2024/07/30 12:30:15 DEB [path camera1] [RTSP source] [s->c] RTSP/1.0 200 OK
> CSeq: 6
> Session: 1
> 
> 
> 2024/07/30 12:30:45 DEB [path camera1] [RTSP source] [c->s] GET_PARAMETER rtsp://localhost:8521/ RTSP/1.0
> Authorization: Digest username="user", realm="SharpRTSPServer", nonce="975155369", uri="rtsp://localhost:8521/", response="260493a353e2989548b13e7248a00a36"
> CSeq: 7
> Session: 1
> User-Agent: gortsplib
> 
> 
> 2024/07/30 12:30:45 DEB [path camera1] [RTSP source] [s->c] RTSP/1.0 200 OK
> CSeq: 7
> Session: 1
> 
> 
> 2024/07/30 12:30:51 INF [RTSP] [conn localhost:65219] closed: terminated
> 2024/07/30 12:30:51 INF [RTSP] [session 828bcc93] destroyed: session timed out__

But when I try to connect FFMPEG to Mediamtx, I don’t get any image, and the only log that differs from correct operation is that it stops at “Decoding VUI”.

How can it stop at that point only when I use Mediamtx if it doesn’t throw any errors and also works correctly if I don’t use the server?

**Comment by aler9** [comment]
Hello, the informations you provided are not enough to replicate the issue. Open a bug report and provide a network dump that allows to see the content of the stream and to spot any difference between the stream generated by the camera and the stream generated by MediaMTX.
